### ISSUE TITLE
[android] no internet access via Wifi for Android 10 devices

### ISSUE THREAD

**Issue by daadu** [alternadom/WiFiFlutter#97]
reported by - @krishnaaro  and @JavaDeveloper08 on #85

**Comment by daadu** [comment]
Will be working on this weekend.

**Comment by daadu** [comment]
New Wifi Infra defines 3 API for different use cases:

<img width="802" alt="Screenshot 2020-11-30 at 6 19 24 PM" src="https://user-images.githubusercontent.com/4963236/100612321-ce76a800-3338-11eb-8130-4036e04da8ad.png">


The package currently uses `Network Request API`, that is why you are not able to connect to internet with the connected Wifi.

A workaround would be to add `bool useInternet` argument to `connect` method, with internally uses `Network Suggestion API`

**Comment by daadu** [comment]
@krishnaaro and @JavaDeveloper08 Can you review #99 PR ?

**Comment by krishnaaro** [comment]
> @krishnaaro and @JavaDeveloper08 Can you review #99 PR ?

Yeah its woking fine and good in Android Q.

`await WiFiForIoTPlugin.removeWifiNetwork('SSID');`
now this throws PlatformException(Error, removeWifiNetwork not supported for Android SDK 29, null)

**Comment by daadu** [comment]
yes it is not supported any more.

**Comment by krishnaaro** [comment]
> yes it is not supported any more.

any way to disconnect that particular "SSID" by any chance .... ?

**Comment by daadu** [comment]
need to use "disconnect" method, in v30+ it will work fine, but for v29 it will only disconnect if the network was connected in same app session.

**Comment by krishnaaro** [comment]
> need to use "disconnect" method

In version < v29 it's happening.

But not disconnecting with v29 even the network was connected in same app session. When i do uninstalling the app then it's disconnecting.

how do i disconnect ... !

**Comment by krishnaaro** [comment]
having trouble in connecting my IOT device(something like Wifi router).

`  wifi_iot:
    git:
      url: https://github.com/daadu/WiFiFlutter.git
      ref: android-major-revamp` this connects my IOT device physically but no connectivity held. 

#99 working good with mobile hotspot connection, but not with IOT device. what happens is, opening and closing WIFI config HTML page very frequently in html viewer (Redmi K20 pro).

**Comment by daadu** [comment]
The device you are connecting to might not be having internet connection

**Comment by krishnaaro** [comment]
disconnect function not happening in Iphone X & +  ..... too

**Comment by krishnaaro** [comment]
for WiFiForIoTPlugin.findAndConnect IOS, 

`flutter: MissingPluginException : MissingPluginException(No implementation found for method setEnabled on channel wifi_iot)
flutter: MissingPluginException : MissingPluginException(No implementation found for method findAndConnect on channel wifi_iot)`

**Comment by luckycreationsindia** [comment]
I see that you remove capability of internet access. Can you add parameter so that we can decide if this feature is required or not!

![Screenshot 2021-01-09 012523](https://user-images.githubusercontent.com/8995339/104058395-a91f6880-5219-11eb-9b92-4fe7c1fddc32.png)

**Comment by luckycreationsindia** [comment]
Also,

I tried this PR,
https://github.com/alternadom/WiFiFlutter/pull/99

If I disable internet access there will be a popup and user can connect to network but once I enable internet access there's no popup nor it connect to selected network.

Note: The network I'm trying to connect has Internet connection

I'm using OnePlus 7 Pro (Android 10)

**Comment by daadu** [comment]
@luckycreationsindia removing `removeCapability` in builder did not solve the problem.

The docs suggest using `NetworkSuggestion API` for connecting to network with internet, as the name suggests they are just suggestions from the app, ultimately the platform decided if to join or not, I suggest you read the docs, see if I am missing something.

For the #99 PR did you approve the suggestion from notification? or it did not appear?

**Comment by daadu** [comment]
This [answere](https://stackoverflow.com/a/61261805/2554745) claims to figure out how to implement it properly. Please feel free to make those suggestions in my fork branch locally and test.

I will only be able to work on this on free weekends.

**Comment by daadu** [comment]
Proper working examples for these new APIs are hard to find on the internet.

**Comment by daadu** [comment]
> @luckycreationsindia removing removeCapability in builder did not solve the problem.

@luckycreationsindia Can you also verify if as well?

**Comment by luckycreationsindia** [comment]
Hi,

I don't know why I'm not getting any notification. I will read the docs and try to find proper way. Also let me try your suggestion and get back to you will results soon.

**Comment by daadu** [comment]
@luckycreationsindia any update?

**Comment by luckycreationsindia** [comment]
> @luckycreationsindia any update?

Hi,

Sorry have been busy with works. Well, I tried but no results.

Here are possibilities to get working internet:

1. Network Suggestions API
2. Guide user to connect manually

But, as you know if user cancel the request from network suggestions API he won't be getting notification again until he reset that.

Suggestion for developers would be add fallback to connect without internet option or guide to connect manually.

**Comment by luckycreationsindia** [comment]
One more thing I modified code to notify user via stream when wifi connects successfully or when that connected wifi gets disconnected.

Reason to add:

1. Added a fix which detects if WiFi gets lost or disconnected manually if might crash the app.

2. Give access to stream to check network connection when user disconnect manually from notification bar or settings.

3. Give flexibility of using stream instead of waiting for result from connect API.
